doc/engineering/coding_standards.md

```diff
@@ -0,0 +1,236 @@
+* All work should be rebased to the originating branch prior to pull request submission.
+
+### Bug Fix Guidelines
+
+Bug fixes must never introduce new features.
+
+Bug fix branches must have originated from an updated version branch in the project which matches the affects version of the bug report.  The pull request should be made against the version branch and all subsequent version branches including master.
+
+All non-bug fix work branches must have originated from an updated master branch in the project and the pull request must be issued against the master branch.
+
+If the pull request was not for the master branch then all subsequent version branches must also have a pull request opened with the bug fix.  It is acceptable to work with 2600Hz to organize a sequence of pull requests when more convenient (such as refactoring the pull request against 3.22 and re-opening it for master).
+
+A test should also be submitted to ensure that the bug is not accidentally reintroduced in the future.
+
+### New Functionality/Features Guidelines
+
+If you are implementing a new feature, also write new test cases.  The primary reason for writing test cases is not to prove that the new feature works correctly, but to make sure that it will be noticed if future changes — perhaps to code that seems unrelated — break the feature.
+
+If you are implementing a new feature, also update the documentation to describe the feature.  If the documentation belongs to a missing document, such as an addition to an undocumented API it is acceptable to only document the addition.  It is nice to create the full missing documentation section ;)
+
+Make sure the patch does not break backward compatibility. In general, we only break backward compatibility in major releases and only for a very good reason and usually after first deprecating the feature one or two releases beforehand.
+
+If you must submit an incompatible change we prefer that the incompatible behaviour require an admin to enable the new functionality via a configuration parameter.
+
+### New Applications Guidelines
+
+TODO
+MR: Thoughts: naming, copyrights section, support contacts that should work, !!!security testing!!!
+
+### Core Updates/Modules Guidelines
+
+TODO
+MR: Thoughts: place where stuff goes (wapi, documents, deps, etc etc and how it should be broken down), comments, tests, support obligations
+
+## Be Ego-less
+
+### Understand and accept that you will make mistakes.
+
+The point is to find them early, before they make it into production. Fortunately, except for the few of us developing rocket guidance software at JPL, mistakes are rarely fatal in our industry, so we can, and should, learn, laugh, and move on.
+
+### You are not your code.
+
+Remember that the entire point of a review is to find problems, and problems will be found. Don't take it personally when one is uncovered.
+
+### No matter how much "karate" you know, someone else will always know more.
```

Comment:
Dunno... @k-anderson why the quotes? The metaphorical karate stands in for one's coding skills?